### PR TITLE
Delete Command Mass Operation

### LINKED ISSUE #31: As a GP clinic management staff, I can group delete patients by category or tag
...so that I can save time when having to delete a large chunk of patients.

### PR DESCRIPTION
**Pull request by Gavino3o**
# Notes:
Resolves #31 

## Expected Behaviour
- `delete` command can now accept multiple indexes instead of only one, to support multiple deletions with a single command.
- As of now, the indexes of the delete command cannot have any duplicate integers. i.e. `delete 1 2 3 4 5 1` should return a useful error message indicating to the user which indexes have been duplicated.
- In the event that both non-positive whole number and duplicated indexes exist in the user input arguments. <br> e.g. `delete 1 1 2 3 a b c`, the parser will prioritise showing an error message for non-positive whole number first.

## Note to other developers
- The "Notes about the command format" section in the User Guide has been updated to include different annotations for `1...n` arguments and `0...n` arguments.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -123,17 +126,18 @@ Examples:
...
+* Deletes all persons at the specified `INDEX…​`.
 * The index refers to the index number shown in the displayed person list.
 * The index **must be a positive integer** 1, 2, 3, …​ with a maximum value of the list size.
+* The indexes must **not** contain any duplicate integers.
 
 Examples:
-* `list -pa` followed by `delete 2` deletes the 2nd patient in the listed patients. 
-* `find -sp n/Betsy` followed by `delete 1` deletes the 1st specialist from the specialists listed in the `find` command.
+* `list -pa` followed by `delete 2` deletes the 2nd patient in the listed patients.
+* `find -sp s/s/Orthopaedic` followed by `delete 2 3 4` deletes the 2nd, 3rd and 4th specialist listed in the `find` command.
```
Comment (vansh284) [inline docs/UserGuide.md:140]: Looks like a typo here

**Review by vansh284 — COMMENTED** [review]
LGTM

File: docs/UserGuide.md (hunk 1 above)
Comment (Gavino3o) [reply docs/UserGuide.md]: Thank you for pointing that out! I have fixed the typo in the User Guide.

**Review by vansh284 — APPROVED** [review]
LGTM

**Comment by Kokseng1** [comment]
LGTM, the checking for duplicates is a nice addition

**Pull request merged**